Formula/util-linux.rb

```diff
@@ -29,20 +29,31 @@ class UtilLinux < Formula
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  conflicts_with "rename", because: "both install `rename` binaries"
+  on_macos do
+    # Change order of includes to avoid "c.h" including macOS system <uuid.h> via <grp.h>.
+    # The missing definitions causes error: use of undeclared identifier 'UUID_STR_LEN'.
+    patch :DATA
+  end
+
+  on_linux do
+    conflicts_with "rename", because: "both install `rename` binaries"
+  end
 
   def install
     args = std_configure_args + %w[
       --disable-silent-rules
     ]
 
     on_macos do
-      args << "--disable-hardlink" # does not build on macOS
       args << "--disable-ipcs" # does not build on macOS
       args << "--disable-ipcrm" # does not build on macOS
       args << "--disable-wall" # already comes with macOS
       args << "--disable-libmount" # does not build on macOS
       args << "--enable-libuuid" # conflicts with ossp-uuid
+
+      # To build `hardlink`, we need to prevent configure from detecting macOS system
+      # <sys/xattr.h>, which doesn't have all expected functions like `lgetxattr`.
+      inreplace "configure", %r{^\tsys/xattr.h \\\n}, ""
```

Comment:
Still trying to find an alternative way to do this. The main goal is to make sure `HAVE_SYS_XATTR_H=0` when compiling `hardlink` to avoid missing functions.

Not familiar enough with autotools to know if there is another way to do this. `configure.ac` uses
```
AC_CHECK_HEADERS([ \
	asm/io.h \
        ...
	sys/xattr.h \
	unistd.h \
	utmp.h \
	utmpx.h \
])
```